src/core/datastructures/tfprimitive.cpp

```diff
@@ -37,6 +37,10 @@ TFPrimitive::TFPrimitive(double pos, const vec4& color) : data_({pos, color}) {}
 
 TFPrimitive::TFPrimitive(const TFPrimitiveData& data) : data_(data) {}
 
+TFPrimitive::TFPrimitive(const TFPrimitive& rhs) : data_(rhs.data_) {}
```

Comment:
Is this not equal to default? Maybe comment on why in that case?

Reply:
necessary because the constructor of Observable is not called!

Reply:
see #199

Reply:
There should really be a comment about that...